relay.go

```diff
@@ -146,12 +144,14 @@ func (r *Relay) CanHop(ctx context.Context, id peer.ID) (bool, error) {
 
 	err = wr.WriteMsg(&msg)
 	if err != nil {
+		s.Reset()
 		return false, err
 	}
-
 	msg.Reset()
 
 	err = rd.ReadMsg(&msg)
+	s.Close()
```

Comment:
there is a defer s.Close(), this isn't necessary.

Reply:
ah, you removed the defered close.

Reply:
(I didn't want to reset on error and then close).